zerver/tests/test_messages.py

```diff
@@ -3164,6 +3166,141 @@ def test_propagate_all_topics(self) -> None:
         self.check_message(id5, topic_name="edited")
         self.check_message(id6, topic_name="topic3")
 
+    def test_move_message_to_stream(self) -> None:
+        user_profile = self.example_user("iago")
+        self.login(user_profile.email)
+        stream = self.make_stream("test move stream")
+        new_stream = self.make_stream("new stream")
+        self.subscribe(user_profile, stream.name)
+        self.subscribe(user_profile, new_stream.name)
+        msg_id = self.send_stream_message(user_profile.email, stream.name,
+                                          topic_name="test", content="First")
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="Second")
+
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="third")
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
+            'message_id': msg_id,
+            'new_stream_id': new_stream.id,
+            'propagate_mode': 'change_all'
+        })
+
+        messages = get_topic_messages(user_profile, stream, "test")
+        self.assertEqual(len(messages), 1)
+
+        messages = get_topic_messages(user_profile, new_stream, "test")
+        self.assertEqual(len(messages), 4)
+
+        self.assert_json_success(result)
+
+    def test_move_message_to_stream_no_allowed(self) -> None:
+        user_profile = self.example_user("aaron")
+        self.login(user_profile.email)
+        stream = self.make_stream("test move stream")
+        new_stream = self.make_stream("new stream")
+        self.subscribe(user_profile, stream.name)
+        self.subscribe(user_profile, new_stream.name)
+        msg_id = self.send_stream_message(user_profile.email, stream.name,
+                                          topic_name="test", content="First")
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="Second")
```

Comment:
This first half of the function seems like it's common to all of these tests; perhaps we should move it to a common helper function that's called in common code?

Reply:
Sure I was trying to do that, but I am getting weird results. Do you have any example of this??

Reply:
There's not much complicated to do, but note that you'll want to name your function not starting with `test_`, or it'll be run as a test itself.